tests/unit/python/fledge/services/core/api/plugins/test_update.py

```diff
@@ -0,0 +1,381 @@
+                                    result = await resp.text()
+                                    response = json.loads(result)
+                                    assert 'id' in response
+                                    assert 'Plugin update started.' == response['message']
+                                    assert response['statusLink'].startswith('fledge/package/update/status?id=')
+                            args, kwargs = insert_tbl_patch.call_args_list[0]
+                            assert 'packages' == args[0]
+                            actual = json.loads(args[1])
+                            assert 'id' in actual
+                            assert pkg_name == actual['name']
+                            assert 'update' == actual['action']
+                            assert -1 == actual['status']
+                            assert '' == actual['log_file_uri']
+                        schedule_patch.assert_called_once_with(svc_name)
+                    plugin_tracked_patch.assert_called_once_with(plugin_type)
+                plugin_installed_patch.assert_called_once_with(plugin_type, False)
+            args, kwargs = query_tbl_patch.call_args_list[0]
+            assert 'packages' == args[0]
+            assert payload == json.loads(args[1])
+
+    async def test_plugin_update_when_in_use(self, client):
+        async def async_mock(return_value):
+            return return_value
+
+        plugin_type = "south"
+        plugin_installed_dirname = "Random"
+        pkg_name = "fledge-{}-random".format(plugin_type)
+        payload = {"return": ["status"], "where": {"column": "action", "condition": "=", "value": "update",
+                                                   "and": {"column": "name", "condition": "=", "value": pkg_name}}}
+        plugin_installed = [{"name": plugin_installed_dirname, "type": plugin_type,
+                             "description": "Random data generation", "version": "1.8.1",
+                             "installedDirectory": "{}/{}".format(plugin_type, plugin_installed_dirname),
+                             "packageName": pkg_name}]
+        insert = {"response": "inserted", "rows_affected": 1}
+        insert_row = {'count': 1, 'rows': [{
+                "id": "c5648940-31ec-4f78-a7a5-b1707e8fe578",
+                "name": plugin_installed_dirname,
+                "action": "update",
+                "status": -1,
+                "log_file_uri": ""
+            }]}
+        svc_name = 'R1'
+        tracked_plugins = [{'plugin': 'sinusoid', 'service': 'S1'}, {'plugin': 'Random', 'service': svc_name}]
+        sch_info = [{'id': '6637c9ff-7090-4774-abca-07dee59a0610', 'enabled': 't'}]
+        server.Server.scheduler = Scheduler(None, None)
+        storage_client_mock = MagicMock(StorageClientAsync)
+        with patch.object(connect, 'get_storage_async', return_value=storage_client_mock):
+            with patch.object(storage_client_mock, 'query_tbl_with_payload',
```

Comment:
are we patching insert? should not insert come from actual code and we patch output of storage?